### PR TITLE
Unify CSV/table-action code; add listings & attendees CSV exports

### PR DESCRIPTION
**Pull request by stefan-burke**
## What & why

Unifies the CSV-export and table-action code into small, reusable, tested modules, then adds CSV exports to two more tables on top of that foundation.

### Styling — consistent table action rows
- New symmetric rows: **`.table-header-actions`** (filters, left-aligned) and **`.table-footer-actions`** (table-wide utilities like CSV export — left on mobile, right on desktop), both margin-free; spacing comes from a shared **`.table-block`** / the existing `.attendees-table-controls` flex container. `.filter-row` is unchanged.
- The type/agent filter's old `.type-filter` becomes `.table-header-actions`, and the near-identical link rendering in the listing-type and logistics-agent filters is pulled into one shared **`renderFilterBar`**.
- The existing **calendar** and **listing-detail** "Export CSV" links are relocated into a `.table-footer-actions` footer below their tables; the **dashboard**, **listings index**, and **attendees browser** gain matching export footers.

### CSV — one module, a generic utility, two new exports
- `src/ui/templates/csv.ts` (281 lines) is split into `src/shared/csv/`: `core.ts` (escape + row join), `attendee-columns.ts`, `attendees.ts`, `calendar.ts`, plus the new `generate.ts` and `listings.ts`.
- **`toCsv(rows, keys)`** — a generic generator: pass typed row objects + an ordered key list; it validates the two against each other (missing/extra/duplicate columns throw) and emits CSV in one pass.
- **`GET /admin/listings/csv`** — first consumer of `toCsv`; supports the same `?type=` filter as the listings views via a shared `filterListingsByType` / `listingTypeFromRequest`.
- **`GET /admin/attendees/csv`** — exports the full filtered set (not just the page), reusing the calendar generator via a shared `toCalendarAttendees` join.

### Docs
- AGENTS.md now records the "always unify duplication, less code is the goal; jscpd 0% is fixed by abstracting, never by dodging detection" rule.

## Checks
`deno task precommit` is green: lint, typecheck, **cpd (0 clones)**, build:edge, and test:coverage (100% line/branch). New unit tests cover `toCsv` and the listings CSV; new route tests cover both export endpoints (including the empty-filter path); template tests cover the relocated/added export links.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01PSNSpug35i1kibPSyZNw3b)_

### CODE REVIEW

File: src/features/admin/dashboard.ts [hunk 1]
```diff
@@ -87,16 +97,28 @@ const handleAdminGet = (request: Request): Promise<Response> =>
...
+  );
+
+/** Handle GET /admin/listings/csv — export every listing (filtered by the same
+ * ?type= category filter the listings views use) as a CSV download. */
+const handleListingsCsvExport: TypedRouteHandler<"GET /admin/listings/csv"> = (
+  request,
+) =>
+  requireSessionOr(request, async () => {
+    const type = listingTypeFromRequest(request);
+    const listings = filterListingsByType(type)(await loadSortedListings());
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/features/admin/dashboard.ts:115]: Good catch. Rather than scope the export to active rows, we've removed the CSV export from the dashboard entirely: its table is active-only and the dashboard isn't the canonical listings view. The export now lives only on `/admin/listings`, which shows active + deactivated listings and so legitimately exports all of them. Fixed in 3ff060a.

---
_Generated by [Claude Code](https://claude.ai/code)_
